### ISSUE TITLE
Got error on `void-variable project-rootfile-plain--`

### ISSUE THREAD

**Issue by stardiviner** [buzztaiki/project-rootfile.el#14]
Here is the error backtrace:

```
Debugger entered--Lisp error: (void-variable project-rootfile-plain--)
  byte-code("\301\302!\210\303\304\305\306\307\302%\210\310\311\312\313\314DD\315\316\317%\210\320\321\322\10!!\323\324\305\325\326##\210\310\327\312\313\330DD\331\316\332%\207" [project-rootfile-plain-- require project custom-declare-group project-rootfile nil "Extension of `project' to detect project root with..." :group custom-declare-variable project-rootfile-list funcall function #f(compiled-function () #<bytecode -0x15f2c88667ac05e1>) "A list of files considered to mark the root of a p..." :type (repeat :type string) cl-defstruct project-rootfile-plain :conc-name "Project backend for plain (outside VCS) project." root :documentation "Root directory of this project." project-rootfile-plain-ignores #f(compiled-function () #<bytecode 0x1f400016fe914>) "List of patterns to add to `project-ignores' for `..." (repeat string)] 7)
  require(project-rootfile nil t)
  (not (require 'project-rootfile nil t))
  (if (not (require 'project-rootfile nil t)) (display-warning 'use-package (format "Cannot load %s" 'project-rootfile) :error) (let ((now (current-time))) (message "%s..." "Configuring package project-rootfile") (prog1 (condition-case err (progn (add-to-list 'project-find-functions #'project-rootfile-try-detect t) (if (member ".project" project-rootfile-list) project-rootfile-list (setq project-rootfile-list (cons ".project" project-rootfile-list))) (if (member ".projectile" project-rootfile-list) project-rootfile-list (setq project-rootfile-list (cons ".projectile" project-rootfile-list))) t) ((debug error) (funcall use-package--warning663 :config err))) (let ((elapsed (float-time (time-subtract ... now)))) (if (> elapsed 0.1) (message "%s...done (%.3fs)" "Configuring package project-rootfile" elapsed) (message "%s...done" "Configuring package project-rootfile"))))))
  (prog1 (if (not (require 'project-rootfile nil t)) (display-warning 'use-package (format "Cannot load %s" 'project-rootfile) :error) (let ((now (current-time))) (message "%s..." "Configuring package project-rootfile") (prog1 (condition-case err (progn (add-to-list 'project-find-functions #'project-rootfile-try-detect t) (if (member ".project" project-rootfile-list) project-rootfile-list (setq project-rootfile-list ...)) (if (member ".projectile" project-rootfile-list) project-rootfile-list (setq project-rootfile-list ...)) t) ((debug error) (funcall use-package--warning663 :config err))) (let ((elapsed (float-time ...))) (if (> elapsed 0.1) (message "%s...done (%.3fs)" "Configuring package project-rootfile" elapsed) (message "%s...done" "Configuring package project-rootfile")))))) (let ((elapsed (float-time (time-subtract (current-time) now)))) (if (> elapsed 0.1) (message "%s...done (%.3fs)" "Loading package project-rootfile" elapsed) (message "%s...done" "Loading package project-rootfile"))))
  (let ((now (current-time))) (message "%s..." "Loading package project-rootfile") (prog1 (if (not (require 'project-rootfile nil t)) (display-warning 'use-package (format "Cannot load %s" 'project-rootfile) :error) (let ((now (current-time))) (message "%s..." "Configuring package project-rootfile") (prog1 (condition-case err (progn (add-to-list ... ... t) (if ... project-rootfile-list ...) (if ... project-rootfile-list ...) t) ((debug error) (funcall use-package--warning663 :config err))) (let ((elapsed ...)) (if (> elapsed 0.1) (message "%s...done (%.3fs)" "Configuring package project-rootfile" elapsed) (message "%s...done" "Configuring package project-rootfile")))))) (let ((elapsed (float-time (time-subtract (current-time) now)))) (if (> elapsed 0.1) (message "%s...done (%.3fs)" "Loading package project-rootfile" elapsed) (message "%s...done" "Loading package project-rootfile")))))
  (condition-case err (let ((now (current-time))) (message "%s..." "Loading package project-rootfile") (prog1 (if (not (require 'project-rootfile nil t)) (display-warning 'use-package (format "Cannot load %s" 'project-rootfile) :error) (let ((now (current-time))) (message "%s..." "Configuring package project-rootfile") (prog1 (condition-case err (progn ... ... ... t) (... ...)) (let (...) (if ... ... ...))))) (let ((elapsed (float-time (time-subtract ... now)))) (if (> elapsed 0.1) (message "%s...done (%.3fs)" "Loading package project-rootfile" elapsed) (message "%s...done" "Loading package project-rootfile"))))) ((debug error) (funcall use-package--warning663 :catch err)))
  (progn (use-package-ensure-elpa 'project-rootfile '(t) 'nil) (defvar use-package--warning663 #'(lambda (keyword err) (let ((msg (format "%s/%s: %s" ... keyword ...))) (display-warning 'use-package msg :error)))) (condition-case err (let ((now (current-time))) (message "%s..." "Loading package project-rootfile") (prog1 (if (not (require 'project-rootfile nil t)) (display-warning 'use-package (format "Cannot load %s" 'project-rootfile) :error) (let ((now ...)) (message "%s..." "Configuring package project-rootfile") (prog1 (condition-case err ... ...) (let ... ...)))) (let ((elapsed (float-time ...))) (if (> elapsed 0.1) (message "%s...done (%.3fs)" "Loading package project-rootfile" elapsed) (message "%s...done" "Loading package project-rootfile"))))) ((debug error) (funcall use-package--warning663 :catch err))))
  eval((progn (use-package-ensure-elpa 'project-rootfile '(t) 'nil) (defvar use-package--warning663 #'(lambda (keyword err) (let ((msg ...)) (display-warning 'use-package msg :error)))) (condition-case err (let ((now (current-time))) (message "%s..." "Loading package project-rootfile") (prog1 (if (not (require ... nil t)) (display-warning 'use-package (format "Cannot load %s" ...) :error) (let (...) (message "%s..." "Configuring package project-rootfile") (prog1 ... ...))) (let ((elapsed ...)) (if (> elapsed 0.1) (message "%s...done (%.3fs)" "Loading package project-rootfile" elapsed) (message "%s...done" "Loading package project-rootfile"))))) ((debug error) (funcall use-package--warning663 :catch err)))) nil)
  elisp--eval-last-sexp(nil)
  eval-last-sexp(nil)
  funcall-interactively(eval-last-sexp nil)
  call-interactively(eval-last-sexp nil nil)
  command-execute(eval-last-sexp)
```

Installed project-rootfile version: `20210305.645`
My Emacs version: `GNU Emacs 28.1 (build 1, aarch64-apple-darwin21.1.0, NS appkit-2113.00 Version 12.0.1 (Build 21A559)) of 2022-05-12`
My System: `macOS Monterey 12.4`

**Comment by stardiviner** [comment]
When I try to re-install this package, I got `*Compile-Log*`:

```

In toplevel form:
project-rootfile.el:93:16: Warning: ‘:conc-name’ called as a function
project-rootfile.el:93:51: Warning: reference to free variable
    ‘project-rootfile-plain--’

In end of data:
project-rootfile.el:130:6: Warning: the function
    ‘project-rootfile-plain--root’ is not known to be defined.
project-rootfile.el:109:23: Warning: the function
    ‘make-project-rootfile-plain’ is not known to be defined.
project-rootfile.el:95:4: Warning: the function ‘root’ is not known to be
    defined.
project-rootfile.el:93:40: Warning: the function ‘:conc-name’ is not known to
    be defined.
project-rootfile.el:93:16: Warning: the function ‘project-rootfile-plain’ is
    not known to be defined.
project-rootfile.el:93:2: Warning: the function ‘cl-defstruct’ might not be
    defined at runtime.
```

**Comment by buzztaiki** [comment]
Ah.. I reproduced same compilation warnings.

```
% emacs -Q -batch -L . -f batch-byte-compile project-rootfile.el

In toplevel form:
project-rootfile.el:91:16: Warning: ‘:conc-name’ called as a function
project-rootfile.el:91:51: Warning: reference to free variable
    ‘project-rootfile-plain--’

In end of data:
project-rootfile.el:128:6: Warning: the function
    ‘project-rootfile-plain--root’ is not known to be defined.
project-rootfile.el:107:23: Warning: the function
    ‘make-project-rootfile-plain’ is not known to be defined.
project-rootfile.el:93:4: Warning: the function ‘root’ is not known to be
    defined.
project-rootfile.el:91:40: Warning: the function ‘:conc-name’ is not known
    to be defined.
project-rootfile.el:91:16: Warning: the function ‘project-rootfile-plain’ is
    not known to be defined.
project-rootfile.el:91:2: Warning: the function ‘cl-defstruct’ might not be
```

https://github.com/buzztaiki/project-rootfile.el/pull/13 will be fix this issue. Please wait a moment.

**Comment by buzztaiki** [comment]
I merged #13.
Thank you for your reporting.
